### PR TITLE
✨ Add windows.optionalfeatures resource

### PR DESCRIPTION
**Pull request by czunker**
Related-to #3705

### CODE REVIEW

**Comment by czunker** [comment]
This allows to show, e.g., the powershell v2 feature on Windows 11:
```
PS C:\Users\User> .\cnquery.exe run -c 'windows.optionalfeatures.where(name == /PowerShell/){*}'
- no provider specified, defaulting to local. Use --help to see all providers.
- no Mondoo configuration file provided, using defaults
windows.optionalfeatures.where: [
  0: {
    displayName: "Windows PowerShell 2.0"
    name: "MicrosoftWindowsPowerShellV2Root"
    enabled: true
    description: "Adds or Removes Windows PowerShell 2.0"
    state: 2
  }
  1: {
    displayName: "Windows PowerShell 2.0 Engine"
    name: "MicrosoftWindowsPowerShellV2"
    enabled: true
    description: "Adds or Removes Windows PowerShell 2.0 Engine"
    state: 2
  }
  2: {
    displayName: "Hyper-V Module for Windows PowerShell"
    name: "Microsoft-Hyper-V-Management-PowerShell"
    enabled: false
    description: "Includes Windows PowerShell cmdlets for managing Hyper-V."
    state: 0
  }
]
```

**Comment by tas50** [comment]
Is the name and output of these the same as the server windows features @czunker ?

File: providers/os/resources/os.lr [hunk 1]
```diff
@@ -1389,6 +1389,9 @@ windows {
 
   // Information about Windows Server roles, role services, and features that are available for installation and installed on a specified server.
   features() []windows.feature
+
+  // Information about optional features in a Windows image.
+  optionalfeatures() []windows.optionalfeature
```
Comment (imilchev) [inline providers/os/resources/os.lr]: nit: why not `optionalFeatures`?

File: providers/os/resources/os.lr [hunk 2]
```diff
@@ -1389,6 +1389,9 @@ windows {
 
   // Information about Windows Server roles, role services, and features that are available for installation and installed on a specified server.
   features() []windows.feature
```
Comment (chris-rock) [inline providers/os/resources/os.lr:1391]: I think we should do:

```
// Information about Windows Server roles, role services, and features that are available for installation and installed on a specified server.
serverFeatures() []windows.serverfeature

// deprecated: use serverFeature
features() []windows.serverfeature
```

**Comment by czunker** [comment]
> Is the name and output of these the same as the server windows features @czunker ?

No and no.

**Comment by czunker** [comment]
```
PS C:\Users\User> .\cnquery.exe run -c "windows.optionalFeature('MicrosoftWindowsPowerShellV2').enabled"
- no provider specified, defaulting to local. Use --help to see all providers.
- no Mondoo configuration file provided, using defaults
[ok] value: true

PS C:\Users\User>
```

File: providers/os/resources/os.lr (hunk 1 above)
Comment (czunker) [reply providers/os/resources/os.lr]: Fixed

File: providers/os/resources/os.lr (hunk 2 above)
Comment (czunker) [reply providers/os/resources/os.lr:1391]: I added the new `serverFeature` and deprecated the old one.

Comment (chris-rock) [reply providers/os/resources/os.lr:1391]: Awesome.

File: providers/os/resources/os.lr [hunk 3]
```diff
@@ -1423,6 +1430,38 @@ windows.feature {
   installState int
 }
 
+// Windows Server feature resource
+windows.serverFeature {
```
Comment (chris-rock) [inline providers/os/resources/os.lr]: I think we should make name the default and be marked private

File: providers/os/resources/os.lr [hunk 4]
```diff
@@ -1406,6 +1412,7 @@ windows.hotfix {
   installedBy string
 }
 
+// Deprecated: use serverFeature instead
 // Windows feature resource
 windows.feature {
```
Comment (chris-rock) [inline providers/os/resources/os.lr]: I think this should be marked private

File: providers/os/resources/os.lr [hunk 5]
```diff
@@ -1423,6 +1430,38 @@ windows.feature {
...
+  // Feature description
+  description string
+  // Whether the feature is installed
+  installed bool
+  // Feature installation state
+  installState int
+}
+
+// Windows optional feature resource
+windows.optionalFeature @defaults("name enabled") {
```
Comment (chris-rock) [inline providers/os/resources/os.lr]: should be marked private

Comment (czunker) [reply providers/os/resources/os.lr]: Fixed.

File: providers/os/resources/os.lr (hunk 4 above)
Comment (czunker) [reply providers/os/resources/os.lr]: Fixed.

File: providers/os/resources/os.lr (hunk 3 above)
Comment (czunker) [reply providers/os/resources/os.lr]: Fixed.

File: providers/os/resources/os.lr.manifest.yaml [hunk 6]
```diff
@@ -1065,6 +1065,10 @@ resources:
       computerInfo: {}
       features: {}
       hotfixes: {}
+      optionalFeatures:
+        min_mondoo_version: latest
```
Comment (chris-rock) [inline providers/os/resources/os.lr.manifest.yaml]:
```suggestion
        min_mondoo_version: 9.0.0
```

File: providers/os/resources/os.lr.manifest.yaml [hunk 7]
```diff
@@ -1158,6 +1162,14 @@ resources:
...
       installedOn: {}
     min_mondoo_version: 5.15.0
+  windows.optionalFeature:
+    fields:
+      description: {}
+      displayName: {}
+      enabled: {}
+      name: {}
+      state: {}
+    min_mondoo_version: latest
```
Comment (chris-rock) [inline providers/os/resources/os.lr.manifest.yaml]:
```suggestion
    min_mondoo_version: 9.0.0
```

File: providers/os/resources/os.lr.manifest.yaml [hunk 8]
```diff
@@ -1183,6 +1195,15 @@ resources:
...
     min_mondoo_version: 5.35.0
+  windows.serverFeature:
+    fields:
+      description: {}
+      displayName: {}
+      installState: {}
+      installed: {}
+      name: {}
+      path: {}
+    min_mondoo_version: latest
```
Comment (chris-rock) [inline providers/os/resources/os.lr.manifest.yaml]:
```suggestion
    min_mondoo_version: 9.0.0
```

File: providers/os/resources/os.lr.manifest.yaml [hunk 9]
```diff
@@ -1065,6 +1065,10 @@ resources:
       computerInfo: {}
       features: {}
       hotfixes: {}
+      optionalFeatures:
+        min_mondoo_version: latest
+      serverFeatures:
+        min_mondoo_version: latest
```
Comment (chris-rock) [inline providers/os/resources/os.lr.manifest.yaml]:
```suggestion
        min_mondoo_version: 9.0.0
```

**Review by chris-rock — APPROVED** [review]
Thank you @czunker

**Pull request merged**